From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: AJ Ferguson <fergusonjva@gmail.com>
Date: Mon, 6 Jan 2025 20:31:00 +1100
Subject: [PATCH] Paper: Fix excess slot updates

Removed since Leaf 1.21.6, added in Paper

Original license: GPLv3
Original project: https://github.com/PaperMC/Paper
Paper pull request: https://github.com/PaperMC/Paper/pull/12654

https://github.com/PaperMC/Paper/commit/e714de636543d61fcd3682705484c8a15f4f3ca6

diff --git a/net/minecraft/world/inventory/AbstractContainerMenu.java b/net/minecraft/world/inventory/AbstractContainerMenu.java
index 9dd3187fd968ab95e9d55b4c8cc74e782cc0f241..46c3353a9722cb9c8073cadbad0ced57bab0c390 100644
--- a/net/minecraft/world/inventory/AbstractContainerMenu.java
+++ b/net/minecraft/world/inventory/AbstractContainerMenu.java
@@ -559,7 +559,7 @@ public abstract class AbstractContainerMenu {
 
                 slot.setChanged();
                 // CraftBukkit start - Make sure the client has the right slot contents
-                if (player instanceof ServerPlayer serverPlayer && slot.getMaxStackSize() != 64) {
+                if (player instanceof ServerPlayer serverPlayer && slot.getMaxStackSize() != Container.MAX_STACK) { // Paper - craftbukkkit - Fix excess slot updates
                     serverPlayer.connection.send(new net.minecraft.network.protocol.game.ClientboundContainerSetSlotPacket(this.containerId, this.incrementStateId(), slot.index, slot.getItem()));
                     // Updating a crafting inventory makes the client reset the result slot, have to send it again
                     if (this.getBukkitView().getType() == org.bukkit.event.inventory.InventoryType.WORKBENCH || this.getBukkitView().getType() == org.bukkit.event.inventory.InventoryType.CRAFTING) {
